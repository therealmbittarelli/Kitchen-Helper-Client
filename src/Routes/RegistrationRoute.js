import React from "react";
import { Link } from "react-router-dom";
import Auth from "../Helpers/Auth";
import "./Registration.css"

export default class Login extends React.Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => { }
    }
  };

  state = {
    error: null
  };

  handleRegistrationSuccess = () => {
    const { history } = this.props;
    history.push("/login");
  };

  state = { error: null };

  createSubmit = ev => {
    ev.preventDefault();
    const { first_name, user_name, password } = ev.target;

    this.setState({ error: null });

    Auth.createAccount({
      first_name: first_name.value,
      user_name: user_name.value,
      password: password.value
    })
      .then(user => {
        first_name.value = "";
        user_name.value = "";
        password.value = "";
        this.handleRegistrationSuccess();
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };

  render() {
    let error = this.state.error;
    return (
      <div className="register">
        <header className="Creation-Header"></header>
        <form className="register-form" onSubmit={this.createSubmit}>
          <div role="alert">
            {error && <p className="red" id="error-message">{this.state.error}</p>}
          </div>
          <label className="field a-field a-field_a2">
            Display name
          </label>
          <input
            className="field__input a-field__input"
            required
            name="first_name"
            placeholder="Name"
          />
          <br />
          <label className="field a-field a-field_a2">
            Username
          </label>
          <input
            className="field__input a-field__input"
            required
            name="user_name"
            placeholder="Username"
          />
          {/* <br />
          <label className="field a-field a-field_a2">
            Email
          </label>
          <input
            className="register-email-input"
            required
            type="email"
            name="user_email"
            placeholder="Email"
          /> */}
          <br />

          <label className="field a-field a-field_a2">
            Password
          </label>
          <input
            className="password-fields"
            required
            name="password"
            type="password"
            placeholder="Password"
          />
          <div className="btn-row">
            <button className="medButton">Create</button>
            <br />

            <Link to="/Login">
              <button className="medButton">Already have an account?</button>
            </Link>
          </div>
        </form>
      </div>
    );
  }
}
