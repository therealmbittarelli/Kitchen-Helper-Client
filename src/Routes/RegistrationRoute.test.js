import React from 'react';
import ReactDOM from 'react-dom';
import RegistrationRoute from './RegistrationRoute';
import { BrowserRouter } from 'react-router-dom';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  let history = {
    push: () => { }
  };
  let location = {
    state: {
      date: '2029-01-22T00:00:00.000Z'
    }
  };
  ReactDOM.render(<BrowserRouter>
    <RegistrationRoute history={history} location={location} />
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  let history = {
    push: () => { }
  };
  let location = {
    state: {
      date: '2029-01-22T00:00:00.000Z'
    }
  };
  const tree = renderer.create(<BrowserRouter>
    <RegistrationRoute history={history} location={location} />
  </BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});