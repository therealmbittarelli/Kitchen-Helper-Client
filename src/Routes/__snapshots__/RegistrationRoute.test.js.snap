// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="register"
>
  <header
    className="Creation-Header"
  />
  <form
    className="register-form"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <label
      className="field a-field a-field_a2"
    >
      Display name
    </label>
    <input
      className="field__input a-field__input"
      name="first_name"
      placeholder="Name"
      required={true}
    />
    <br />
    <label
      className="field a-field a-field_a2"
    >
      Username
    </label>
    <input
      className="field__input a-field__input"
      name="user_name"
      placeholder="Username"
      required={true}
    />
    <br />
    <label
      className="field a-field a-field_a2"
    >
      Password
    </label>
    <input
      className="password-fields"
      name="password"
      placeholder="Password"
      required={true}
      type="password"
    />
    <div
      className="btn-row"
    >
      <button
        className="medButton"
      >
        Create
      </button>
      <br />
      <a
        href="/Login"
        onClick={[Function]}
      >
        <button
          className="medButton"
        >
          Already have an account?
        </button>
      </a>
    </div>
  </form>
</div>
`;
